patches/node/build_add_gn_build_files.patch

```diff
@@ -449,6 +450,234 @@ index 0000000000000000000000000000000000000000..a47875642d8f825c84ba1e82e3892a97
++  }
++
++  if (target_cpu == "ia32" || target_cpu == "x64" || target_cpu == "x32") {
++    defines += [
++      "HAVE_SSSE3=1",
++      "HAVE_SSE41=1",
++      "HAVE_SSE42=1",
++      "HAVE_AVX=1",
++      "HAVE_AVX2=1",
++    ]
++
++    deps += [
++      ":base64_ssse3",
++      ":base64_sse41",
++      ":base64_sse42",
++      ":base64_avx",
++      ":base64_avx2",
++    ]
++  } else {
++    sources += [
++      "base64/lib/arch/ssse3/ssse3_codec.c",
++      "base64/lib/arch/sse41/sse41_codec.c",
++      "base64/lib/arch/sse42/sse42_codec.c",
++      "base64/lib/arch/avx/avx_codec.c",
++      "base64/lib/arch/avx2/avx2_codec.c",
++    ]
++  }
++}
++
++static_library("base64_ssse3") {
++  include_dirs = [
++    "base64/include",
++    "base64/lib"
++  ]
++
++  defines = [
++    "BASE64_STATIC_DEFINE",
++    "HAVE_SSSE3=1"
++  ]
++
++  sources = [ "base64/lib/arch/ssse3/ssse3_codec.c" ]
++
++  if (!is_win) {
++    cflags = ["-mssse3"]
++  }
++}
++
++
++static_library("base64_sse41") {
++  include_dirs = [
++    "base64/include",
++    "base64/lib"
++  ]
++
++  defines = [
++    "BASE64_STATIC_DEFINE",
++    "HAVE_SSE41=1"
++  ]
++
++  sources = [ "base64/lib/arch/sse41/sse41_codec.c" ]
++
++  if (!is_win) {
++    cflags = ["-msse4.1"]
++  }
++}
++
++static_library("base64_sse42") {
++  include_dirs = [
++    "base64/include",
++    "base64/lib"
++  ]
++
++  defines = [
++    "BASE64_STATIC_DEFINE",
++    "HAVE_SSE42=1"
++  ]
++
++  sources = [ "base64/lib/arch/sse42/sse42_codec.c" ]
++
++  if (!is_win) {
++    cflags = ["-msse4.2"]
++  }
++}
++
++static_library("base64_avx") {
++  include_dirs = [
++    "base64/include",
++    "base64/lib"
++  ]
++
++  defines = [
++    "BASE64_STATIC_DEFINE",
++    "HAVE_AVX=1"
++  ]
++
++  sources = [ "base64/lib/arch/avx/avx_codec.c" ]
++
++  if (!is_win) {
++    cflags = ["-mavx"]
++  }
++}
++
++static_library("base64_avx2") {
++  include_dirs = [
++    "base64/include",
++    "base64/lib"
++  ]
++
++  defines = [
++    "BASE64_STATIC_DEFINE",
++    "HAVE_AVX2=1"
++  ]
++
++  sources = [ "base64/lib/arch/avx2/avx2_codec.c" ]
++
++  if (!is_win) {
++    cflags = ["-mavx2"]
++  }
++}
++
++static_library("base64_neon32") {
++  include_dirs = [
++    "base64/include",
++    "base64/lib"
++  ]
++
++  defines = [
++    "BASE64_STATIC_DEFINE",
++    "HAVE_NEON32=1"
++  ]
++
++  cflags_c = [ "-Wno-implicit-fallthrough" ]
++
++  sources = [ "base64/lib/arch/neon32/neon32_codec.c" ]
++}
++
++static_library("base64_neon64") {
++  include_dirs = [
++    "base64/include",
++    "base64/lib"
++  ]
++
++  defines = [
++    "BASE64_STATIC_DEFINE",
++    "HAVE_NEON64=1"
++  ]
++
++  cflags_c = [ "-Wno-implicit-fallthrough" ]
++
++  sources = [ "base64/lib/arch/neon64/neon64_codec.c" ]
++}
+diff --git a/deps/base64/base64/lib/arch/avx/codec.c b/deps/base64/base64/lib/arch/avx/avx_codec.c
+similarity index 100%
+rename from deps/base64/base64/lib/arch/avx/codec.c
+rename to deps/base64/base64/lib/arch/avx/avx_codec.c
```

Comment:
gn doesn't seem to allow duplicate filenames even if they're in different directories - this is what happens otherwise:

```
electron on git:upgrade-nodejs-v18 ❯ e build                               10:12AM
Running "e load-xcode --quiet"
Running "python3 goma_auth.py info" in /Users/codebytere/.electron_build_tools/third_party/goma
Running "ninja -j 200 electron" in /Users/codebytere/Developer/electron-gn/src/out/Testing
[0/1] Regenerating ninja files
ERROR at //third_party/electron_node/deps/base64/BUILD.gn:10:1: Duplicate object file
static_library("base64") {
^-------------------------
The target //third_party/electron_node/deps/base64:base64
generates two object files with the same name:
  obj/third_party/electron_node/deps/base64/base64/codec.o

It could be you accidentally have a file listed twice in the
sources. Or, depending on how your toolchain maps sources to
object files, two source files with the same name in different
directories could map to the same object file.

In the latter case, either rename one of the files or move one of
the sources to a separate source_set to avoid them both being in
the same target.
FAILED: build.ninja.stamp
```

Reply:
Generally the way to solve this in GN is to use separate targets. Make an `base64_avx` target or something that has the avx-specific files in it. object names only need to be unique within a single target.